lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1699,9 +1700,16 @@ def _get_deployable_vhd_path(
         matches = SAS_URL_PATTERN.match(vhd_path)
         if not matches:
             return vhd_path
-        log.debug("found the vhd is a sas url, it needs to be copied...")
+        log.debug("found the vhd is a sas url, it may need to be copied.")
 
+        # get original vhd's hash key for comparing.
+        original_key: Optional[bytearray] = None
         original_vhd_path = vhd_path
+        original_blob_client = BlobClient.from_blob_url(original_vhd_path)
```

Comment:
does it work when the VHD is not in the same subscription as the test subscription?

Reply:
Yes, it must be a sas URL.